cocli/COMID.md

```diff
@@ -0,0 +1,129 @@
+
+# CoMIDs manipulation
+
+The `comid` subcommand allows you to create, display and validate CoMIDs.
+
+## Create
+
+Use the `comid create` subcommand to create a CBOR-encoded CoMID, passing its
+JSON representation via the `--template` switch (or equivalently its `-t` shorthand):
+
+* Please inspect `comid` JSON templates as examples under `data/comid/templates` `comid-*.json`
+
+```
+$ cocli comid create --template data/comid/templates/comid-dice-refval.json
+```
+On success, you should see something like the following printed to stdout:
+```
+>> created "comid-dice-refval.cbor" from "comid-dice-refval.json"
+```
+
+The CBOR-encoded CoMID file is stored in the current working directory with a
+name derived from its template.  If you want, you can specify a different
+target directory using the `--output-dir` command line switch (abbrev. `-o`)
+```
+$ cocli comid create --template data/comid/templates/comid-dice-refval.json --output-dir /tmp
+>> created "/tmp/comid-dice-refval.cbor" from "comid-dice-refval.json"
+```
+Note that the output directory, as well as all its parent directories, MUST
+pre-exist.
+
+You can also create multiple CoMIDs in one go.  Suppose all your templates are
+stored in the `templates/` folder:
+```
+$ tree templates/
+templates/
+├── comid-dice-refval1.json
+├── comid-dice-refval2.json
+...
+└── comid-dice-refvaln.json
+```
+Then, you can use the `--template-dir` (abbrev. `-T`), and let the tool load,
+validate, and CBOR-encode the templates one by one:
+```
+$ cocli comid create --template-dir templates
+>> created "comid-dice-refval1.cbor" from "templates/comid-dice-refval1.json"
+>> created "comid-dice-refval2.cbor" from "templates/comid-dice-refval2.json"
+...
+>> created "comid-dice-refvaln.cbor" from "templates/comid-dice-refvaln.json"
+```
+
+You can specify both the `-T` and `-t` switches as many times as needed, and
+even combine them in one invocation:
+```
+$ cocli comid create -T comid-templates/ \
+                   -T comid-templates-aux/ \
+                   -t extra-comid.json \
+                   -t yet-another-comid.json \
+                   -o /var/spool/comid
+```
+
+**NOTE** that since the output file name is deterministically generated from the
+template file name, all the template files (when from different directories)
+MUST have different base names.
+
+
+## Display
+
+Use the `comid display` subcommand to print to stdout one or more CBOR-encoded
+CoMIDs in human readable (JSON) format.
+
+You can supply individual files using the `--file` switch (abbrev. `-f`), or
+directories that may (or may not) contain CoMID files using the `--dir` switch
+(abbrev. `-d`).  Only valid CoMIDs will be displayed, and any decoding or
+validation error will be printed alongside the corresponding file name.
+
+For example:
+```
+$ cocli comid display --file data/comid/comid-dice-refval.cbor
```

Comment:
this didn't work for me:

```sh
➜  cocli git:(tidy-documentation) ✗ ./cocli comid display --file data/comid/comid-dice-refval.cbor
>> failed displaying "data/comid/comid-dice-refval.cbor": CBOR decoding failed: error unmarshalling field "Triples": error unmarshalling field "ReferenceValues": error unmarshalling field "Flags": expected map (CBOR Major Type 5), found Major Type 0
```

Reply:
Do you have the right version of `cocli` ?

Reply:
I don't know what the "right" version is.  I just compiled it from sources and using it to test the examples.

Reply:
Can you please try with:
go install github.com/veraison/corim/cocli@latest

Reply:
I could, but @latest points to v1.1.1 which is 8 months old, and many changes have gotten into the codebase since then.

Reply:
We cannot use latest for the `tip` at the moment, hence I retained @latest for now!

Reply:
Once we migrate to the `tip` then all the samples will need to be udpated, I will raise a `seperate` issue for that!

Reply:
The problem is deeper than that.  Imagine someone creating their CoRIMs using `cocli@latest`, submitting them to veraison services (which links `corim@v1.1.2-0.20230912171018-eeb7bd486d3c`) and getting back a decode error...

Reply:
This is the same state as of today!

Reply:
and that's not good

Reply:
Yes, agree we need to do it at some point in time: See issue #103 created to address this issue!